engine/src/main/scala/cromwell/engine/workflow/ValidateActor.scala

```diff
@@ -56,67 +55,80 @@ class ValidateActor()
     * 2.) Raw inputs and it's coercion
     * 3.) Workflow Options
     * 4.) RuntimeAttributes w.r.t. a backend
+    *
     * @return The result of the validation as instance of `ValidationResult`
     */
-  private def validateAll(wdlSource: WdlSource, workflowInputs: Option[WdlJson], workflowOptions: Option[WdlJson]): Future[ValidationResult] = {
-    (for {
-      namespaceWithWorkflow <- Future(NamespaceWithWorkflow.load(wdlSource))
-      validatedInputs <- Future(validateInputs(workflowInputs, namespaceWithWorkflow))
-      validatedWorkflowOptions <- Future(validateWorkflowOptions(workflowOptions, CromwellBackend.backend()))
-      validatedRuntimeAttrs <- Future(validateRuntimeAttributes(namespaceWithWorkflow))
-    } yield ValidationSuccess(namespaceWithWorkflow, validatedInputs, validatedWorkflowOptions, validatedRuntimeAttrs)) recover {
-      case reason => ValidationFailure(reason)
+  private def validateAll(wdlSource: WdlSource, workflowInputs: Option[WdlJson], workflowOptions: Option[WdlJson]): Future[ValidationResult] = Future {
+    val namespace = validateNamespace(wdlSource)
+    val result = namespace match {
+      case scalaz.Success(ns) =>
+        val options = validateWorkflowOptions(CromwellBackend.backend(), workflowOptions: Option[WdlJson])
+        val inputs = validateInputs(ns, workflowInputs)
+        val runtimeAttributes = validateRuntimeAttributes(ns)
+        (options |@| inputs |@| runtimeAttributes) ((opt, in, runAttr) => ValidationSuccess(ns, Option(in), Option(opt), runAttr))
+
+      case scalaz.Failure(failure) => failure.failure
+    }
+
+    result match {
+      case scalaz.Success(success) => success
+      case scalaz.Failure(failure) =>
```

Comment:
This variable should be called something plural like `failures` since it's a `NonEmptyList[String]`.